### PR TITLE
Edm 922

### PR DESCRIPTION
**Pull request by romfreiman**
test2

## Summary by CodeRabbit

- **New Features**
  - Introduced a new GitHub Action for collecting and uploading logs from Kubernetes pods.
  - Added new steps in the end-to-end and smoke testing workflows to collect and upload logs after tests.

- **Configuration Enhancements**
  - Expanded the Flight Control API configuration with new HTTP timeout and limit parameters.
  - Added support for custom duration handling in configuration settings.

- **Middleware Improvements**
  - Enhanced request handling by adding middleware for request size and URL length limitations.

- **Bug Fixes**
  - Updated server configuration to improve error handling and resource management.

### CODE REVIEW

File: deploy/helm/flightctl/templates/flightctl-api-config.yaml [hunk 1]
```diff
@@ -17,6 +17,13 @@ data:
         address: :3443
         agentEndpointAddress: :7443
         agentGrpcAddress: :7444
+        httpReadTimeout: {{ .Values.api.httpReadTimeout | default "5m" | quote }}
+        httpReadHeaderTimeout: {{ .Values.api.httpReadHeaderTimeout | default "5m" | quote }}
+        httpWriteTimeout: {{ .Values.api.httpWriteTimeout | default "5m" | quote }}
```
[the remark this answers is not in the document]
Comment (avishayt) [reply deploy/helm/flightctl/templates/flightctl-api-config.yaml:22]: We set it higher than usual to avoid gRPC timeouts

**Comment by romfreiman** [comment]
@coderabbitai create a test plan

**Comment by romfreiman** [comment]
@greptile-apps review

**Comment by romfreiman** [comment]
@greptile-apps greptile-review performance security

**Comment by romfreiman** [comment]
@greptileai review greptile-review performance security

**Pull request closed without merging**